addons/survey/static/src/js/survey_form.js

```diff
@@ -74,70 +70,235 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend({
         }
     },
 
-    /*
-    * When clicking on a breadcrumb enabled item, redirect to the target page.
-    * Uses the submit flow to validate and save the answer before going to the specified page.
-    *
-    * @private
-    * @param {Event} event
-    */
-    _onBreadcrumbClick: function (event) {
+    _onSubmit: function (event) {
         event.preventDefault();
-        this._submitForm({'previous_page_id': $(event.currentTarget).data('previousPageId')});
+        this._submitForm($(event.currentTarget));
     },
 
-    _onSubmit: function (event) {
-        event.preventDefault();
-        if ($(event.currentTarget).val() === 'previous') {
-            this._submitForm({'previous_page_id': $(event.currentTarget).data('previousPageId')});
-        } else {
-            this._submitForm({});
-        }
+    // Custom Events
+    // -------------------------------------------------------------------------
+
+    _onBreadcrumbClick: function (event) {
+        this._submitForm(event.data);
     },
 
-    _submitForm: function (params) {
+    // SUBMIT
+    // -------------------------------------------------------------------------
+
+    /**
+    * This function will send a json rpc call to the server to
+    * - start the survey (if we are on start screen)
+    * - submit the answers of the current page
+    * Before submitting the answers, they are first validated to avoid latency from the server
+    * and allow a fade out/fade in transition of the next question.
+    * Depending of the target, the next page will be
+    * - For 'Continue' button: the next page
+    * - For 'Back' button or Breadcrumb items: the page with the given 'previous_page_id'.
+    */
+    _submitForm: function ($target) {
         var self = this;
-        var $form = this.$('form');
-        var formData = new FormData($form[0]);
+        var params = {};
+        var route = "/survey/submit";
 
-        this._prepareSubmitValues(formData, params);
+        if (this.options.isStartScreen) {
+            route = "/survey/begin";
+        } else {
+            if ($target.hasClass('breadcrumb-item')) {
+                params = {'previous_page_id': $target.data('pageId')};
+            } else {
+                params = $target.val() === 'previous' ? {'previous_page_id': $target.data('previousPageId')} : {};
+            }
+            var $form = this.$('form');
+            var formData = new FormData($form[0]);
 
-        this._resetErrors();
+            if ($target.hasClass('o_survey_timer')) {
+                $target.val('finish');
```

Comment:
If I am not wrong, it indicates this is the last "submit" button of survey.

Instead of a target value, couldn't we be a bit more explicit and use a data on the button, like data-form-ended ?

Reply:
Changed to an "options" param on _submitForm.